pkg/plugins/resources/hcl/target.go

```diff
@@ -0,0 +1,90 @@
+package hcl
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/minamijoyo/hcledit/editor"
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/pipeline/scm"
+	"github.com/updatecli/updatecli/pkg/core/result"
+)
+
+func (h *Hcl) Target(source string, scm scm.ScmHandler, dryRun bool, resultTarget *result.Target) error {
+	if scm != nil {
+		h.UpdateAbsoluteFilePath(scm.GetDirectory())
+	}
+
+	for _, f := range h.files {
+		resourceFile := f
+
+		// Target doesn't support updating files on remote http location
+		if strings.HasPrefix(resourceFile.filePath, "https://") ||
+			strings.HasPrefix(resourceFile.filePath, "http://") {
+			return fmt.Errorf("%s URL scheme is not supported for HCL target: %q", result.FAILURE, h.spec.File)
+		}
+	}
+
+	if err := h.Read(); err != nil {
+		return fmt.Errorf("reading hcl file: %w", err)
+	}
+
+	query := h.spec.Key
+
+	valueToWrite := source
+	if h.spec.Value != "" {
+		valueToWrite = h.spec.Value
+		logrus.Info("INFO: Using spec.Value instead of source input value.")
```

Comment:
I think we could use the debug level such as 
```suggestion
		logrus.Debugln("Using spec.Value instead of source input value.")
```